app/models/otu.rb

```diff
@@ -364,6 +273,114 @@ def self.select_optimized(user_id, project_id, target = nil)
     h
   end
 
+  def current_collection_objects
+    collection_objects.where(taxon_determinations: {position: 1})
+  end
+
+  # @return [Boolean]
+  #   whether or not this otu is coordinate (see coordinate_otus) with this otu
+  def coordinate_with?(otu_id)
+    Otu.coordinate_otus(otu_id).where(otus: {id: id}).any?
+  end
+
+  # TODO: Deprecate for helper method, HTML does not belong here
+  def otu_name
+    if !name.blank?
+      name
+    elsif !taxon_name_id.nil?
+      taxon_name.cached_html_name_and_author_year
+    else
+      nil
+    end
+  end
+
+  # TODO: move to helper method likely
+  def distribution_geoJSON
+    a_ds = Gis::GeoJSON.feature_collection(geographic_areas_from_asserted_distributions, :asserted_distributions)
+    c_os = Gis::GeoJSON.feature_collection(collecting_events, :collecting_events_georeferences)
+    c_es = Gis::GeoJSON.feature_collection(geographic_areas_from_collecting_events, :collecting_events_geographic_area)
+    Gis::GeoJSON.aggregation([a_ds, c_os, c_es], :distribution)
+  end
+
+  # TODO: need's spec
+  # A convienence method to wrap coordinate_otus and descendant_of_taxon_name
+  # @return Scope
+  def coordinate_otus_with_children
+    if taxon_name_id.nil?
+      Otu.coordinate_otus(id)
+    else
+      Otu.descendant_of_taxon_name(taxon_name.valid_taxon_name.id)     
+    end
+  end
+
+  # @return [Array]
+  #   of ancestral otu_ids
+  # !! This method does not fork, as soon as 2 ancestors are
+  # !! hit the list terminates.
+  def ancestor_otu_ids(prefer_unlabelled_otus: true)
+    ids =  []
+    a = parent_otu_id(prefer_unlabelled_otus: true)
+    while a
+      ids.push a
+      b = Otu.find(a)
+      a = b.parent_otu_id(prefer_unlabelled_otus: true)
+    end
+    ids
+  end
+
+  # @return [Array]
+  #   all bilogical associations this Otu is part of
+  def all_biological_associations
+    # !! If self relationships are ever made possible this needs a DISTINCT clause
+    BiologicalAssociation.find_by_sql(
```

Comment:
@mjy would this be equivalent or does `UNION` has something specific we need?
```ruby
BiologicalAssociation.where(biological_association_object: self).or(
  BiologicalAssociation.where(biological_association_subject: self)
)

Reply:
I believe that's OK in this case.  Earlier I had experimented with some performance vs. UNION, IIRC that was with taxon name relationships, and found a massive gain in that case, but I can't remember if that case was identical to this.  The other thing that tickles my brain is that UNION makes the result DISTINCT, and having to call DISTINCT when scopes were chained might have lead to issues.  It's most likely though that I just used an old pattern before the `or()` syntax was available.

Reply:
Resulting query is this:
```ruby
SELECT "biological_associations".*
FROM "biological_associations" 
WHERE (
  "biological_associations"."biological_association_object_type" = $1 AND
  "biological_associations"."biological_association_object_id" = $2 OR
  "biological_associations"."biological_association_subject_type" = $3 AND
  "biological_associations"."biological_association_subject_id" = $4
)
[
  ["biological_association_object_type", "Otu"],
  ["biological_association_object_id", 1],
  ["biological_association_subject_type", "Otu"],
  ["biological_association_subject_id", 1]
]
```

In this case `UNION` is not doing much work with respect to query above since it is including `id` so in essence it is already as `DISTINCT` as `UNION` query is.

Reply:
I'm fine to change it, let's just make sure there is a unit test behind the code, there may already be.